classes/Tools.php

```diff
@@ -3051,31 +3049,28 @@ public static function getDirectories($path)
     /**
      * Return the directory list from the given $path using php glob function.
      *
-     * @param string $path
-     *
-     * @return array
+     * @return array<string>
      */
-    public static function getDirectoriesWithGlob($path)
+    public static function getDirectoriesWithGlob(string $path): array
     {
         $directoryList = glob($path . '/*', GLOB_ONLYDIR | GLOB_NOSORT);
-        array_walk(
-            $directoryList,
-            function (&$absolutePath, $key) {
-                $absolutePath = substr($absolutePath, strrpos($absolutePath, '/') + 1);
-            }
-        );
+        if ($directoryList === false) {
+            $directoryList = [];
+        }
+
+        $directoryList = array_map(function ($path) {
+            return substr($path, strrpos($path, '/') + 1);
+        }, $directoryList);
 
         return $directoryList;
     }
 
     /**
      * Return the directory list from the given $path using php readdir function.
      *
-     * @param string $path
-     *
-     * @return array
+     * @return array<string>
      */
-    public static function getDirectoriesWithReaddir($path)
+    public static function getDirectoriesWithReaddir(string $path): array
```

Comment:
We  can't change signatures of public functions in minor version

Reply:
done